### PR TITLE
Iss2698 wsl pipe

### PR DESCRIPTION
**Pull request by tamuhey**
- From #2698 
- If we are in WSL, avoid to use `xonsh_preexec_fn` when "pipe"
  - add `ON_WSL` attribute in [platform.py](https://github.com/xonsh/xonsh/blob/master/xonsh/platform.py)
  - If `ON_WSL` is `True`,  xonsh_preexec_fn = no_pg_xonsh_preexec_fn when pipe.

- I am not sure if it is proper solution.

### CODE REVIEW

File: news/iss2698_wsl_pipe.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+**Deprecated:**
+
+
+**Removed:**
+
+
+**Fixed:**
+
+
+**Security:**
```
Comment (scopatz) [inline news/iss2698_wsl_pipe.rst]: I think to get the tests to be happy, you can remove the unused categories

**Comment by scopatz** [comment]
Thanks for putting this in @tamuhey! Let us know if you need any help!

**Comment by tamuhey** [comment]
I ran pytest inside the docker, which is run with `xonsh-in-docker.py`.
I found one test failure, as follows.

> ============================= test session starts ==============================
>  platform linux -- Python 3.4.9, pytest-4.0.2, py-1.7.0, pluggy-0.8.0
>  rootdir: /xonsh, inifile: setup.cfg
>  plugins: xonsh-0.8.5, timeout-1.3.3, flake8-1.0.2, cov-2.6.0
>  collected 117 items
> 
>  test_builtins.py ..........................................F............ [ 47%]
>  ..............................................................           [100%]
> 
>  =================================== FAILURES ===================================
>  ____________________________ test_repath_backslash _____________________________
> 
>  home_env = <module 'builtins' (built-in)>
> 
>      @skip_if_on_windows
>      def test_repath_backslash(home_env):
>          exp = os.listdir(HOME_PATH)
>          exp = {p for p in exp if re.match(r"\w\w.*", p)}
>          exp = {os.path.join(HOME_PATH, p) for p in exp}
>          obs = set(pathsearch(regexsearch, r"~/\w\w.*"))
>  >       assert exp == obs
>  E       AssertionError: assert set([]) == set(['~/\\w\\w.*'])
>  E         Extra items in the right set:
>  E         '~/\\w\\w.*'
>  E         Use -v to get the full diff
> 
>  test_builtins.py:60: AssertionError
>  ===================== 1 failed, 116 passed in 1.96 seconds =====================

**Comment by tamuhey** [comment]
The test also fails in master branch.

**Comment by scopatz** [comment]
updated from master

**Comment by scopatz** [comment]
Thanks @tamuhey!

**Comment by scopatz** [comment]
Sorry this took me so long to get to!

**Pull request merged**